### ISSUE TITLE
[BUG] MPerClassSampler dictionary not working for Torch Tensors

### ISSUE THREAD

**Issue by JoOkuma** [KevinMusgrave/pytorch-metric-learning#12]
The MPerClassSampler does not stratify the batch correctly when provided with a torch tensor of labels, it works correctly with numpy arrays.

Code for bug reproduction:
```python
import torch as th
from pytorch_metric_learning import samplers

labels = th.LongTensor([0, 1, 2, 0, 0, 1, 1, 2, 2])
sampler = samplers.MPerClassSampler(labels, m=2)
print(sampler.labels_to_indices)
```

Output:
```
defaultdict(<class 'list'>, {tensor(0): array([0]), tensor(1): array([1]), tensor(2): array([2]), tensor(0): array([3]), tensor(0): array([4]), tensor(1): array([5]), tensor(1): array([6]), tensor(2): array([7]), tensor(2): array([8])})
```

Expected Output:
```
defaultdict(<class 'list'>, {0: array([0, 3, 4]), 1: array([1, 5, 6]), 2: array([2, 7, 8])})
```

**Comment by KevinMusgrave** [comment]
Good point. It's kind of funny my datasets have always returned numpy arrays so I never had this problem.

**Comment by KevinMusgrave** [comment]
I'll fix this in the next version.

**Comment by JoOkuma** [comment]
Do you think it would be a good idea to add a few assertions on the library?

**Comment by KevinMusgrave** [comment]
That could be a good idea! Were you thinking of something in particular?

**Comment by JoOkuma** [comment]
> That could be a good idea! Were you thinking of something in particular?

I'm not completely familiar with this library, but I think two functions `validateLabelArray` and `validadeDataArray` to check, convert or raise an error on the user inputs would be enough for most of the function calls.

**Comment by KevinMusgrave** [comment]
Makes sense
